source/simulator/free_surface.cc

```diff
@@ -57,9 +57,20 @@ namespace aspect
                                 std_cxx11::_1,
                                 // discard pressure_scaling,
                                 // discard rebuild_stokes_matrix,
-                                // discard scratch object,
+                                std_cxx11::_4,
                                 std_cxx11::_5));
 
+    // Note that we do not want face_material_model_data, because we do not
+    // connect to a face assembler. We instead connect to a normal assembler,
+    // and compute our own material_model_inputs in apply_stabilization
+    // (because we want to use the solution instead of the current_linearization_point
+    // to compute the material properties).
+    sim.assemblers->stokes_system_assembler_on_boundary_face_properties.need_face_material_model_data = false;
```

Comment:
I am not sure this is correct. You are setting it unconditionally to false here, so what if some other code needs the material data?